File: php/findNumbers.php
<?php

class Solution {

     private function findNumberLength($num) {
        $length = 1;

        while($num >= 10) {
            $num = ($num -  ($num % 10)) / 10;
            
            $length++;
        }
        return $length;
    }

    private function isEven($num) {
        return $this->findNumberLength($num) % 2 === 0;
    }

    /**
     * @param Integer[] $nums
     * @return Integer
     */
    function findNumbers($nums) {
        $evenCounts = 0;
        for ($i=0; $i < count($nums); $i++) { 
            $isNumberEven = $this->isEven($nums[$i]);
            if ($isNumberEven) {
                $evenCounts++;
            }
        }

        return $evenCounts;
    }
}

$nums = [100000];

$solution = new Solution();

echo $solution->findNumbers($nums);

File: 2025/leetcode /findNumbers.py
class Solution:
    def divideBy(self, num: int, count: int) -> int:
        if num < 10:
            return count
        count += 1
        return self.divideBy(num/10, count)
    
    def findNumbers(self, nums) -> int:
        even = 0
        for num in nums:
          count = 0
          while num >= 10:
            num = num / 10
            count += 1
          
          if count > 0 and count % 2 == 1:
            even += 1

          # evenCount = self.divideBy(num, count=0)
          # if evenCount%2 == 1:
          #   even += 1
        return even

solution = Solution()

print(solution.findNumbers([100000]))